src/utils/delaySending.js

```diff
@@ -0,0 +1,26 @@
+const { promisify } = require('util');
+
+const sleep = promisify(setTimeout);
+
+/**
+ *
+ * This function will send out one request per second, until all requests have
+ * been sent.
+ *
+ * @param {Function} sender function that needs to be delayed
+ * @param {Array} items an array of items to iterate over and process through the sender
+ * @param {Number} ms a time in miliseconds that each request should wait for before processing
+ */
+const delaySending = async function delaySending(sender, items, ms) {
+  items.map(async (item, i) => {
```

Comment:
(minor) If we're not capturing the value I suppose `forEach` is fine? Unless you want to make this blocking, then use `map` with `Promise.all` - or if you want it serial rather than parallel, then we could use regular `for` loop with nested `await`s.

```suggestion
  items.forEach(async (item, i) => {
```

Reply:
Good call out. We want a serial block here to avoid rate limits.